### DISCUSSION TITLE
自动分析代码和翻译注释工具

### DISCUSSION THREAD

**Discussion by huihuang-chen** [CN-annotation-team/redis7.0-chinese-annotated#75] [Ideas]
有没有可能我们做一个软件自动分析出源码流程来，
比如给一个方法，通过对对应语言的编译器做二次开发，自动把这个方法的所有方法调用栈给分析出来，然后在页面上显示方法调用栈中每个方法的代码，注释，这里注释也可以通过用谷歌翻译做一次翻译在展示

**Comment by fengxiaop** [comment]
想法很不错

**Comment by huihuang-chen** [comment]
java的我应该可以做，java的前端编译javac和反编译javap我都很熟，平时对于工作里面一些比较简单的功能代码，我可以用我自己改造后的javac来自动生成的，而且也能分析出所有和前端对接的api，感觉做代码流程分析工具是没什么问题，主要是对于一些动态代理的代码，和第三方jar包插入方法调用栈可能不太好分析，这个需要做反编译之后分析，会麻烦点，不过也能做

**Comment by artikell** [comment]
有个想法是，通过LLVM工具链统一编译到IR，然后，所有语言都对IR进行分析。但是，之前没有这么尝试过。
